lib/attack.go

```diff
@@ -14,24 +14,71 @@ import (
 	"sync"
 	"time"
 
+	"github.com/gorilla/mux"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
 	"golang.org/x/net/http2"
 )
 
 // Attacker is an attack executor which wraps an http.Client
 type Attacker struct {
-	dialer     *net.Dialer
-	client     http.Client
-	stopch     chan struct{}
-	workers    uint64
-	maxWorkers uint64
-	maxBody    int64
-	redirects  int
-	seqmu      sync.Mutex
-	seq        uint64
-	began      time.Time
-	chunked    bool
+	dialer      *net.Dialer
+	client      http.Client
+	stopch      chan struct{}
+	workers     uint64
+	maxWorkers  uint64
+	maxBody     int64
+	redirects   int
+	seqmu       sync.Mutex
+	seq         uint64
+	began       time.Time
+	chunked     bool
+	promEnabled bool
+	promBind    string
+	promPort    int
+	promPath    string
+	promSrv     *http.Server
 }
 
+//Prometheus metrics
+var requestSecondsHistogram = promauto.NewHistogramVec(prometheus.HistogramOpts{
+	Name:    "request_seconds",
+	Help:    "Request latency",
+	Buckets: []float64{0.1, 0.2, 0.5, 1.0, 2.0, 5.0, 10.0, 20, 50},
```

Comment:
How have you chosen these buckets?

Reply:
Based on my experience. Do you have any other references? Surely we can change.